src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -1078,7 +1085,10 @@ def prepare_create(self, daemon_spec: CephadmDaemonDeploySpec) -> CephadmDaemonD
                         f"ssl_endpoint={build_url(host=daemon_spec.ip, port=port).lstrip('/')}")
                 else:
                     args.append(f"ssl_port={port}")
-                args.append(f"ssl_certificate=config://rgw/cert/{spec.service_name()}")
+                if not spec.generate_cert:
+                    args.append(f"ssl_certificate=config://rgw/cert/{spec.service_name()}")
+                else:
+                    args.append(f"ssl_certificate=config://rgw/cert/{daemon_spec.name()}")
```

Comment:
nit: maybe we should switch the order (to avoid the `not` check) unless we want to force the order for a very specific reason

Reply:
swapped the order

Reply:
the reason for the different name is because in the generate_cert case we're doing a per daemon cert, while in the other case it's a service wide cert. We need to use the daemon name for the name in the former case so there's no name conflict between the certs for each RGW daemon